File: app/src/main/java/com/culqi/MainActivity.java
package com.culqi;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

import com.culqilib.model.Card;
import com.culqilib.Token;
import com.culqilib.listener.TokenCallback;
import com.culqilib.model.Error;

import org.json.JSONObject;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final TextView textView = findViewById(R.id.text_view);

        Card card = new Card("4111111111111111","123",9,2020,"annonimo79@gmail.com");

        Token token = new Token("pk_test_Jf2vS2zIlimqIgm8");

        token.createToken(getApplicationContext(), card, new TokenCallback() {
            @Override
            public void onSuccess(JSONObject token) {
                try {
                    textView.setText(token.get("id").toString());
                } catch (Exception ex){
                    textView.setText(ex.getMessage());
                }
            }

            @Override
            public void onError(Error error) {
                textView.setText(error.getUserMessage());
            }
        });
    }
}

File: app/src/main/java/com/culqi/MainActivity.kt
package com.culqi

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View

import com.culqilib.Token
import com.culqilib.listener.TokenCallback
import com.culqilib.model.TokenError
import com.culqilib.model.TokenSuccess
import com.culqilib.restServices.deserializer.TokenDeserializer
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val card = TokenDeserializer.CardRequest(cardNumber="4111111111111111",
                                                    cvv="123",
                                                    expirationMonth=9,
                                                    expirationYear=2020,
                                                    email="annonimo79@gmail.com")

        val token = Token("pk_test_Jf2vS2zIlimqIgm8")
        token.createToken(card, object : TokenCallback {
            override fun onSuccess(token: TokenSuccess) {
                textView.text = token.id

                textView.visibility = View.VISIBLE
                progressBar.visibility = View.GONE
            }

            override fun onError(tokenError: TokenError) {
                textView.text = if (tokenError.userMessage.isNotEmpty()) tokenError.userMessage else tokenError.merchantMessage
                Log.e("MerchantMessage", tokenError.merchantMessage)
                Log.e("UserMessage", tokenError.userMessage)
                Log.e("Type", tokenError.type)

                textView.visibility = View.VISIBLE
                progressBar.visibility = View.GONE
            }
        })
    }
}
